### PR TITLE
Scripts: ci: testplan: check exists _test_plan_partial file before open

### PR DESCRIPTION
**Pull request by congnguyenhuu**
Incase _test_plan_partial.json is not generated, be error if not check exists this file. Need check file before open.
Missing module gitpython when run manually test_plan. Add module gitpython in requirements for test plan scripts.

### CODE REVIEW

File: scripts/ci/test_plan.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 # SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 NXP
```
Comment (nashif) [inline scripts/ci/test_plan.py:3]: you need (c) here

File: scripts/ci/test_plan.py [hunk 2]
```diff
@@ -110,11 +111,11 @@ def get_plan(self, options, integration=False):
 
         logging.info(" ".join(cmd))
         _ = subprocess.call(cmd)
-        with open(fname, newline='') as jsonfile:
-            json_data = json.load(jsonfile)
-            suites = json_data.get("testsuites", [])
-            self.all_tests.extend(suites)
         if os.path.exists(fname):
```
Comment (nashif) [inline scripts/ci/test_plan.py:114]: what are the cases where this file is not available?

Comment (congnguyenhuu) [reply scripts/ci/test_plan.py:114]: When we run tests plan for a board thật is not supported

Comment (nashif) [reply scripts/ci/test_plan.py:114]: ok, but your patch will just silently skip errors and continue, this is not what we want, the failure here is correct, I mean we can make it nicer, check for the file and abort with an error message, but what you are doing above, will jus continue despite the failure AFAIK

Comment (congnguyenhuu) [reply scripts/ci/test_plan.py:114]: yeah, I can add an error message

File: scripts/ci/test_plan.py (hunk 1 above)
Comment (congnguyenhuu) [reply scripts/ci/test_plan.py:3]: because (c) means "Copyright", so NXP is using it

Comment (manuargue) [reply scripts/ci/test_plan.py:3]: the rationales it that `(c)` means "Copyright", and having "Copyright (c)" is redundant, so the preferred way according to the company policy is to use "Copyright <year> NXP" (see for instance https://github.com/zephyrproject-rtos/hal_nxp/blob/master/dts/nxp/nxp_imx/mimx6sx-pinctrl.dtsi#L2). Older copyrights from NXP may include "Copyright (c)".

File: scripts/ci/test_plan.py (hunk 2 above)
Comment (manuargue) [reply scripts/ci/test_plan.py:114]: There’s actually an error message already saying “unrecognized platform”, before the exception traceback. So perhaps `subprocess.call` could be replaced with `subprocess.check_call` to capture the exit status of invoking twister and abort early if that fails.

**Comment by congnguyenhuu** [comment]
The error message is actually unnecessary. Because it only exist  when we run correct boards that are supported. 
I will close PR.

**Pull request closed without merging**